### PR TITLE
ref(py3): Remove all `six.PYx` references

### PR DESCRIPTION
**Pull request by BYK**

### CODE REVIEW

**Review by joshuarli — APPROVED** [review]

File: src/sentry/data_export/tasks.py [hunk 1]
```diff
@@ -234,27 +231,13 @@ def process_rows(processor, data_export, batch_size, offset):
 
 @handle_snuba_errors(logger)
 def process_issues_by_tag(processor, limit, offset):
-    gtv_list_unicode = processor.get_serialized_data(limit=limit, offset=offset)
-    # TODO(python3): Remove next block once the 'csv' module has been updated
-    # to Python 3
-    if six.PY2:
-        gtv_list = convert_to_utf8(gtv_list_unicode)
```
Comment (joshuarli) [inline src/sentry/data_export/tasks.py:241]: Looks like you can remove `convert_to_utf8` completely.

File: src/sentry/web/frontend/mixins/csv.py [hunk 2]
```diff
@@ -7,18 +7,9 @@
...
-    def encode_row(row):
-        return row
-
-
-else:
-
-    def encode_row(row):
-        return map(force_bytes, row)
+def encode_row(row):
+    return row
```
Comment (joshuarli) [inline src/sentry/web/frontend/mixins/csv.py:12]: Let's inline this.

**Review by joshuarli — APPROVED** [review]
Looks good assuming comments are addressed.

Also looks like a bunch of pre-commit things to address.

File: src/sentry/data_export/tasks.py [hunk 3]
```diff
@@ -115,10 +115,7 @@ def assemble_download(
                 # module is able to do, it will NOT write bytes like in py2).
                 # Because of this we use the codec getwriter to transform our
                 # file handle to a stream writer that will encode to utf8.
-                if six.PY2:
-                    tfw = tf
-                else:
-                    tfw = codecs.getwriter("utf-8")(tf)
+                tfw = codecs.getwriter("utf-8")(tf)
```
Comment (EvanPurkhiser) [inline src/sentry/data_export/tasks.py:118]: We should clean up the comment above too.

Actually I'm pretty sure this whole bit of code becomes a lot simpler if we open the file in text mode as utf-8.

So maybe actually we should leave this for now until we can clean up?

File: src/sentry/data_export/tasks.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/data_export/tasks.py:241]: Yeah this whole module can be refactored a bit to be simpler

File: src/sentry/utils/compat/__init__.py [hunk 4]
```diff
@@ -34,58 +34,34 @@ def _identity(x):
...
+#
+# > Changed in version 3.0: The result is always unsigned. To generate the
+# > same numeric value across all Python versions and platforms, use
+# > crc32(data) & 0xffffffff.
+#
+# However this will not work when transitioning between versions, as the
+# value MUST match what was generated in python 2.
+#
+# We can sign the return value using the following bit math to ensure we
+# match the python2 output of crc32.
```
Comment (EvanPurkhiser) [inline src/sentry/utils/compat/__init__.py:57]: Would be good to update this comment now to reflect why this is now here for historical reasons.

File: tests/sentry/utils/hashlib/tests.py [hunk 5]
```diff
@@ -32,10 +32,7 @@ def test_hash_values(seed, value, hash):
 
 
 def test_hashvalues_python23_strings():
```
Comment (EvanPurkhiser) [inline tests/sentry/utils/hashlib/tests.py:34]: Just delete the test lol

File: tests/acceptance/test_emails.py [hunk 6]
```diff
@@ -37,17 +37,9 @@
...
-    # NOTE that we _cannot_ just set the version of the seed, as there are more
-    # differences. See [0].
-    #
-    # [0]: https://stackoverflow.com/questions/55647936/random-randint-shows-different-output-in-python-2-x-and-python-3-x-with-same-see/55648073
-    version_suffix = "_py2" if six.PY2 else ""
-
     # "sso unlinked without password"
     # => "sso_unlinked_without_password.txt"
-    filename = name.replace(" ", "_") + version_suffix + ".txt"
+    filename = name.replace(" ", "_") + ".txt"
```
Comment (EvanPurkhiser) [inline tests/acceptance/test_emails.py:42]: Gotta delete all these test fixtures now too

Comment (joshuarli) [reply tests/acceptance/test_emails.py:42]: Ah that's a good catch.

File: src/sentry/data_export/tasks.py (hunk 3 above)
Comment (BYK) [reply src/sentry/data_export/tasks.py:118]: I'd still remove the `PY2` check as it is irrelevant. For the rest, I'll leave the comment

Comment (EvanPurkhiser) [reply src/sentry/data_export/tasks.py:118]: Yeah, please leave some TODO(py3) type comments :)

**Review by joshuarli — APPROVED** [review]

**Comment by BYK** [comment]
No idea why the test is failing, will investigate.

**Pull request merged**